### PR TITLE
Added Samples for Kotlin.text last() functions

### PR DESCRIPTION
**Pull request by zekehul**
Added samples for Kotlin.text `last()` and `last(predicate)` functions
Issue: https://youtrack.jetbrains.com/issue/KT-20357

### CODE REVIEW

**Review by ilya-g — CHANGES_REQUESTED** [review]
Please reformat the new sample according to the Kotlin style guide (`Ctrl/Cmd+Alt+L` shortcut can do it).

**Comment by qurbonzoda** [comment]
You can run gradle task `./gradlew :kotlin-stdlib:samples:check` to make sure your sample compiles and actually works as expected.

File: libraries/stdlib/samples/test/samples/text/char.kt [hunk 1]
```diff
@@ -124,4 +124,20 @@ class Chars {
         assertTrue('a'.equals('A', true))
     }
 
-}
\ No newline at end of file
+    @sample
```
Comment (qurbonzoda) [inline libraries/stdlib/samples/test/samples/text/char.kt:127]: @**S**ample

File: libraries/stdlib/samples/test/samples/text/char.kt [hunk 2]
```diff
@@ -124,4 +124,20 @@ class Chars {
...
+    fun last() {
+	    val chars = listOf('a', 'z', '1', '2')
```
Comment (qurbonzoda) [inline libraries/stdlib/samples/test/samples/text/char.kt:129]: We use 4 spaces indentation

**Comment by qurbonzoda** [comment]
The `last` and `last(predicate)` functions are not related to `Char`.
It would be very useful to provide samples for `String.last()` and `String.last(predicate)`. For example:
```
    @Sample
    fun last() {
        val string = "Kotlin 1.4"
        assertEquals('4', string.last())
        assertEquals('n', string.last { it.isLetter() })
    }
```

File: libraries/stdlib/samples/test/samples/text/char.kt (hunk 1 above)
Comment (zekehul) [reply libraries/stdlib/samples/test/samples/text/char.kt]: Fixed

File: libraries/stdlib/samples/test/samples/text/char.kt (hunk 2 above)
Comment (zekehul) [reply libraries/stdlib/samples/test/samples/text/char.kt]: This should be fixed as well

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 3]
```diff
@@ -408,4 +408,18 @@ class Strings {
         assertPrints(matchDetails(inputString, toFind, 2), "Searching for 'ever' in 'Never ever give up' starting at position 2: Found at 6")
         assertPrints(matchDetails(inputString, toFind, 10), "Searching for 'ever' in 'Never ever give up' starting at position 10: Not found")
     }
+
+    @Sample
+    fun last() {
+        val str0 = "Sample String 0"
+        val str1 = ""
+        assertEquals('0', str0.last())
+        assertEquals('g', str0.last( {it.isLetter()} ))
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt:417]: - Use `assertPrints(actualValue, expectedString)` instead of `assertEquals`
- The trailing lambda should be formatted as `str0.last { it.isLetter() }`

File: libraries/stdlib/samples/test/samples/text/strings.kt [hunk 4]
```diff
@@ -408,4 +408,18 @@ class Strings {
...
+        assertFailsWith<NoSuchElementException>{
+            str1.last()
+        }
+        assertFailsWith<NoSuchElementException>{
+            str0.last( {it > 'z'} )
+        }
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/strings.kt:423]: Please place the entire `assertFailsWith` call on the single line, otherwise it won't be correctly formatted after a conversion to a runnable sample.

**Review by ilya-g — CHANGES_REQUESTED** [review]

File: libraries/stdlib/samples/test/samples/text/char.kt [hunk 5]
```diff
@@ -123,5 +123,4 @@ class Chars {
         assertFalse('a'.equals('A', false))
         assertTrue('a'.equals('A', true))
     }
-
-}
\ No newline at end of file
+}
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/text/char.kt:126]: Better to revert unrelated changes

**Comment by qurbonzoda** [comment]
The notes were fixed and changes were merged manually.
Thanks for the effort!
